File: corpus/cs1/1281-ida-a-freira/solutions/a-optimal/solve.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <iomanip>

using namespace std;

void solve() {
    int n;
    if (!(cin >> n)) return;

    while (n--) {
        int m;
        cin >> m;
        unordered_map<string, double> catalog;
        for (int i = 0; i < m; ++i) {
            string name;
            double price;
            cin >> name >> price;
            catalog[name] = price;
        }

        int p;
        cin >> p;
        double total = 0.0;
        for (int i = 0; i < p; ++i) {
            string name;
            int qty;
            cin >> name >> qty;
            if (catalog.count(name)) {
                total += catalog[name] * qty;
            }
        }

        cout << fixed << setprecision(2) << "R$ " << total << endl;
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    solve();
    return 0;
}

File: corpus/cs1/1281-ida-a-freira/solutions/a-optimal/solve.py
import sys

def solve():
    try:
        line = sys.stdin.readline()
        if not line:
            return
        n = int(line.strip())
        
        for _ in range(n):
            m = int(sys.stdin.readline().strip())
            catalog = {}
            for _ in range(m):
                # Using rsplit to handle possible spaces in names if allowed 
                # (though spec says name then price)
                parts = sys.stdin.readline().strip().split()
                if len(parts) >= 2:
                    price = float(parts[-1])
                    name = " ".join(parts[:-1])
                    catalog[name] = price
            
            p = int(sys.stdin.readline().strip())
            total = 0.0
            for _ in range(p):
                parts = sys.stdin.readline().strip().split()
                if len(parts) >= 2:
                    qty = int(parts[-1])
                    name = " ".join(parts[:-1])
                    if name in catalog:
                        total += catalog[name] * qty
            
            print(f"R$ {total:.2f}")
    except EOFError:
        pass
    except ValueError:
        pass

if __name__ == "__main__":
    solve()
